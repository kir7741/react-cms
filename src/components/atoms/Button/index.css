.button {
	border: 1px solid transparent;
	padding: 6px 12px;
	border-radius: 5px;
	cursor: pointer;
}

.default-btn {
	background-color: transparent;
	transition: all 0.5s;

	&:hover {
		background-color: #e6e6e6;
	}
}
