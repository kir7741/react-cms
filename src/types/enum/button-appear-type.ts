// eslint-disable-next-line no-shadow
export enum ButtonAppearType {
	DEFAULT = 'default',
	PRIMARY = 'primary'
}
