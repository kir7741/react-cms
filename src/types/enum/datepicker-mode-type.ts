/**
 * datepick 選擇模式列舉
 *
 * @export
 * @enum {number}
 */
// eslint-disable-next-line no-shadow
export enum DatepickerModeType {

	/**
	 * 年份
	 */
	YEAR = '1',

	/**
	 * 月份
	 */
	MONTH = '2',

	/**
	 * 日期
	 */
	DATE = '3',

}
