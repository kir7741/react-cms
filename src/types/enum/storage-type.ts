// eslint-disable-next-line no-shadow
export enum StorageType {

	SESSION,

	LOCAL

}
