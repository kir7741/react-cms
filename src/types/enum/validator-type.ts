/**
 * 檢核種類的列舉
 *
 * @export
 * @enum {number}
 */
// eslint-disable-next-line no-shadow
export enum ValidatorType {

	/**
	 * 必填
	 */
	REQUIRED = 'required'

}
