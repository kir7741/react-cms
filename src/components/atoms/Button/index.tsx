import React, { InputHTMLAttributes } from 'react';
import classnames from 'classnames';

import { ButtonAppearType } from '../../../types/enum/button-appear-type.enum';

import styles from './index.css';

interface ButtonProperty extends InputHTMLAttributes<HTMLButtonElement> {
	className?: string;
	text: string;
	type: 'button' | 'submit';
	appear?: ButtonAppearType;
	onClick?: () => void;
}

const Button: React.FC<ButtonProperty> = ({
	className,
	text,
	type,
	disabled = false,
	appear = ButtonAppearType.DEFAULT,
	children,
	onClick
}) => (
	<button
	  // eslint-disable-next-line react/button-has-type
		type={type}
		disabled={disabled}
		className={classnames(styles.button, className, {
			[styles.defaultBtn]: appear === ButtonAppearType.DEFAULT
		})}
		onClick={onClick}
	>
		{text}
		{children}
	</button>
);

export default Button;
