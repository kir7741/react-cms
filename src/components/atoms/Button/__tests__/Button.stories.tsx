import React from 'react';
import { storiesOf } from '@storybook/react';

import Button from 'components/atoms/Button';
import { ButtonAppearType } from '../../../../types/enum/button-appear-type.enum';

const stories = storiesOf('atoms/Button', module);

stories.add('__interactive', () => (
	<Button type="submit" text="登入" appear={ButtonAppearType.DEFAULT}/>
));
