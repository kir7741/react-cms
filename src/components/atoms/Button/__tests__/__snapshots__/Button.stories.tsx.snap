// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <button
    class="button defaultBtn"
    type="submit"
  >
    登入
  </button>
</DocumentFragment>
`;
